Library/Model/SPIPayAtTable.m

```diff
@@ -19,57 +19,64 @@ @implementation SPIBillStatusResponse
     if (_billData == nil || [_billData isEqualToString:@""]) {
         return [[NSArray alloc] init];
     }
-    
+
     NSData *decodedData = [[NSData alloc] initWithBase64EncodedString:_billData options:0];
     NSString *decodedString = [[NSString alloc] initWithData:decodedData encoding:NSUTF8StringEncoding];
     SPILog(@"Decoded payment history: %@", decodedString);
-   
+
     NSError *jsonError = nil;
-    NSData *jsonData  = [NSJSONSerialization dataWithJSONObject:decodedString options:0 error:&jsonError];
-    
+    NSDictionary *jsonData  = [NSJSONSerialization JSONObjectWithData:decodedData options:NSJSONReadingAllowFragments error:&jsonError];
+
     if (jsonError) {
         SPILog(@"ERROR: Payment history decoding error: %@", jsonError);
         return [[NSArray alloc] init];
     }
-    
-    NSMutableArray * paymentHistory = [[NSMutableArray alloc] init];
-    NSDictionary *jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&jsonError];
-    for (NSDictionary * historyValue in jsonObject.allValues.firstObject) {
+
+    NSMutableArray *paymentHistory = [[NSMutableArray alloc] init];
+    for (NSDictionary * historyValue in jsonData) {
         SPIPaymentHistoryEntry *historyItem = [[SPIPaymentHistoryEntry alloc] initWithDictionary:historyValue];
         [paymentHistory addObject:historyItem];
     }
-    
+
     return [paymentHistory copy];
 }
 
+
 + (NSString *)toBillData:(NSArray<SPIPaymentHistoryEntry *> *)ph {
     if (ph.count < 1) {
         return @"";
     }
-    
+
     NSError *writeError = nil;
     NSData *jsonData = [NSJSONSerialization dataWithJSONObject:ph options:NSJSONWritingPrettyPrinted error:&writeError];
     return [jsonData base64EncodedStringWithOptions:0];
 }
 
 - (SPIMessage *)toMessage:(NSString *)messageId {
     NSMutableDictionary *data = [[NSMutableDictionary alloc] init];
-    [data setObject:[NSNumber numberWithInteger:BillRetrievalResultSuccess] forKey:@"success"];
+    BOOL success = _result == BillRetrievalResultSuccess;
+    [data setValue:[NSNumber numberWithBool:success] forKey:@"success"];
     if (_billId.length > 0) {
-        [data setObject:_billId forKey:@"bill_id"];
+        [data setValue:_billId forKey:@"bill_id"];
     }
     if (_tableId.length > 0) {
-        [data setObject:_tableId forKey:@"table_id"];
+        [data setValue:_tableId forKey:@"table_id"];
     }
     if (_result == BillRetrievalResultSuccess) {
-        [data setObject:@"bill_total_amount" forKey:[NSNumber numberWithInteger:_totalAmount]];
-        [data setObject:@"bill_outstanding_amount" forKey:[NSNumber numberWithInteger:_outstandingAmount]];
-        [data setObject:@"bill_payment_history" forKey:[self getBillPaymentHistory]];
+        [data setValue:[NSNumber numberWithInteger:_totalAmount] forKey:@"bill_total_amount"];
```

Comment:
Should work if you want to remember for next time, no big issue if you dont do this for this version: ```suggestion
        [data setValue:@_totalAmount forKey:@"bill_total_amount"];
```

Reply:
it gives the error "Unexpected ''@" in program " when I change it like your suggestion

Reply:
Done